xLights/effects/SketchPanel.cpp

```diff
@@ -86,12 +86,11 @@ SketchPanel::SketchPanel(wxWindow* parent, wxWindowID id /*=wxID_ANY*/, const wx
     TextCtrl_SketchDef->SetEditable(false);
 
     auto bgLabel = new wxStaticText(this, wxID_ANY, "Background:");
-    FilePicker_SketchBackground = new wxFilePickerCtrl(this, ID_FILEPICKER_SketchBackground,
-                                                       wxEmptyString, imgSelect, imgFilters,
-                                                       wxDefaultPosition, wxDefaultSize,
-                                                       wxFLP_FILE_MUST_EXIST | wxFLP_OPEN | wxFLP_USE_TEXTCTRL,
-                                                       wxDefaultValidator, "ID_FILEPICKER_SketchBackground");
-    //FilePicker_SketchBackground->GetTextCtrl()->SetEditable(false);
+    FilePicker_SketchBackground = new xlSketchFilePickerCtrl(this, ID_FILEPICKER_SketchBackground,
```

Comment:
you need to add the custom class to the wxsmith file

Reply:
I see PicturesPanel.wxs which this would be a clone of.  Where is it used and how can I test this part of the code?
More to learn ..

Reply:
Yeah I'm pretty sure Kevin coded up the SketchPanel manually so there is no WXS file for it.  We usually do initial dialog/panel design in CodeBlocks which generates the wxs file.  For files that do have a wxs if you make changes inside the special quoted areas they will get lost next time another dev regenerates the file in Codeblocks.